### ISSUE TITLE
Unable to train with charges by specifying charges_weight

### ISSUE THREAD

**Issue by kamtung-tonybill** [ACEsuit/mace#998]
I was attempting to train a MACE model with energy, forces, and charges with the following input parameters:

```
mace_run_train \
    --name="MACE_model" \
    --train_file="../recalculated_converged_data_with_charges.xyz" \
    --valid_fraction=0.05 \
    --config_type_weights='{"Default":1.0}' \
    --E0s='{1:-13.66651632894863, 7:-266.359522973356, 8:-432.1205561049978}' \
    --model="ScaleShiftMACE" \
    --hidden_irreps='128x0e + 128x1o' \
    --r_max=6.0 \
    --batch_size=33 \
    --valid_batch_size=33\
    --max_num_epochs=2000 \
    --patience=30 \
    --swa \
    --start_swa=1800 \
    --energy_key="DFT_energy" \
    --forces_key="DFT_forces" \
    --charges_key="initial_charges" \
    --energy_weight=1.0 \
    --forces_weight=1.0 \
    --charges_weight=1.0 \
    --ema \
    --ema_decay=0.99 \
    --amsgrad \
    --restart_latest \
    --device=cuda \
    --scheduler_patience=15\
    --restart_latest\
    --clip=2.0\
    --default_dtype=float32\
    --keep_checkpoints\
    --enable_cueq=True
```

However, even though the atomic_data.py file implemented the charges_weight, I was not able to proceed to train a model with charges. I got the following error:

usage: mace_run_train [-h] [--config CONFIG] --name NAME [--seed SEED] [--work_dir WORK_DIR] [--log_dir LOG_DIR] [--model_dir MODEL_DIR] [--checkpoints_dir CHECKPOINTS_DIR] [--results_dir RESULTS_DIR]
                      [--downloads_dir DOWNLOADS_DIR] [--device {cpu,cuda,mps,xpu}] [--default_dtype {float32,float64}] [--distributed] [--log_level LOG_LEVEL] [--plot PLOT] [--plot_frequency PLOT_FREQUENCY]
                      [--error_table {PerAtomRMSE,TotalRMSE,PerAtomRMSEstressvirials,PerAtomMAEstressvirials,PerAtomMAE,TotalMAE,DipoleRMSE,DipoleMAE,EnergyDipoleRMSE}]
                      [--model {BOTNet,MACE,ScaleShiftMACE,ScaleShiftBOTNet,AtomicDipolesMACE,EnergyDipolesMACE}] [--r_max R_MAX] [--radial_type {bessel,gaussian,chebyshev}] [--num_radial_basis NUM_RADIAL_BASIS]
                      [--num_cutoff_basis NUM_CUTOFF_BASIS] [--pair_repulsion] [--distance_transform {None,Agnesi,Soft}]
                      [--interaction {RealAgnosticResidualInteractionBlock,RealAgnosticAttResidualInteractionBlock,RealAgnosticInteractionBlock,RealAgnosticDensityInteractionBlock,RealAgnosticDensityResidualInteractionBlock}]
                      [--interaction_first {RealAgnosticResidualInteractionBlock,RealAgnosticInteractionBlock,RealAgnosticDensityInteractionBlock,RealAgnosticDensityResidualInteractionBlock}] [--max_ell MAX_ELL]
                      [--correlation CORRELATION] [--num_interactions NUM_INTERACTIONS] [--MLP_irreps MLP_IRREPS] [--radial_MLP RADIAL_MLP] [--hidden_irreps HIDDEN_IRREPS] [--num_channels NUM_CHANNELS] [--max_L MAX_L]
                      [--gate {silu,tanh,abs,None}] [--scaling {std_scaling,rms_forces_scaling,no_scaling}] [--avg_num_neighbors AVG_NUM_NEIGHBORS] [--compute_avg_num_neighbors COMPUTE_AVG_NUM_NEIGHBORS]
                      [--compute_stress COMPUTE_STRESS] [--compute_forces COMPUTE_FORCES] [--train_file TRAIN_FILE] [--valid_file VALID_FILE] [--valid_fraction VALID_FRACTION] [--test_file TEST_FILE]
                      [--test_dir TEST_DIR] [--multi_processed_test MULTI_PROCESSED_TEST] [--num_workers NUM_WORKERS] [--pin_memory PIN_MEMORY] [--atomic_numbers ATOMIC_NUMBERS] [--mean MEAN] [--std STD]
                      [--statistics_file STATISTICS_FILE] [--E0s E0S] [--foundation_filter_elements FOUNDATION_FILTER_ELEMENTS] [--heads HEADS] [--multiheads_finetuning MULTIHEADS_FINETUNING]
                      [--foundation_head FOUNDATION_HEAD] [--weight_pt_head WEIGHT_PT_HEAD] [--num_samples_pt NUM_SAMPLES_PT] [--force_mh_ft_lr FORCE_MH_FT_LR] [--subselect_pt {fps,random}]
                      [--filter_type_pt {none,combinations,inclusive,exclusive}] [--pt_train_file PT_TRAIN_FILE] [--pt_valid_file PT_VALID_FILE] [--foundation_model_elements FOUNDATION_MODEL_ELEMENTS]
                      [--keep_isolated_atoms KEEP_ISOLATED_ATOMS] [--energy_key ENERGY_KEY] [--forces_key FORCES_KEY] [--virials_key VIRIALS_KEY] [--stress_key STRESS_KEY] [--dipole_key DIPOLE_KEY]
                      [--head_key HEAD_KEY] [--charges_key CHARGES_KEY] [--skip_evaluate_heads SKIP_EVALUATE_HEADS]
                      [--loss {ef,weighted,forces_only,virials,stress,dipole,huber,universal,energy_forces_dipole,l1l2energyforces}] [--forces_weight FORCES_WEIGHT] [--swa_forces_weight SWA_FORCES_WEIGHT]
                      [--energy_weight ENERGY_WEIGHT] [--swa_energy_weight SWA_ENERGY_WEIGHT] [--virials_weight VIRIALS_WEIGHT] [--swa_virials_weight SWA_VIRIALS_WEIGHT] [--stress_weight STRESS_WEIGHT]
                      [--swa_stress_weight SWA_STRESS_WEIGHT] [--dipole_weight DIPOLE_WEIGHT] [--swa_dipole_weight SWA_DIPOLE_WEIGHT] [--config_type_weights CONFIG_TYPE_WEIGHTS] [--huber_delta HUBER_DELTA]
                      [--optimizer {adam,adamw,schedulefree}] [--beta BETA] [--batch_size BATCH_SIZE] [--valid_batch_size VALID_BATCH_SIZE] [--lr LR] [--swa_lr SWA_LR] [--weight_decay WEIGHT_DECAY] [--amsgrad]
                      [--scheduler SCHEDULER] [--lr_factor LR_FACTOR] [--scheduler_patience SCHEDULER_PATIENCE] [--lr_scheduler_gamma LR_SCHEDULER_GAMMA] [--swa] [--start_swa START_SWA] [--lbfgs] [--ema]
                      [--ema_decay EMA_DECAY] [--max_num_epochs MAX_NUM_EPOCHS] [--patience PATIENCE] [--foundation_model FOUNDATION_MODEL] [--foundation_model_readout] [--eval_interval EVAL_INTERVAL]
                      [--keep_checkpoints] [--save_all_checkpoints] [--restart_latest] [--save_cpu] [--clip_grad CLIP_GRAD] [--dry_run] [--enable_cueq ENABLE_CUEQ] [--wandb] [--wandb_dir WANDB_DIR]
                      [--wandb_project WANDB_PROJECT] [--wandb_entity WANDB_ENTITY] [--wandb_name WANDB_NAME] [--wandb_log_hypers WANDB_LOG_HYPERS]
mace_run_train: error: unrecognized arguments: --charges_weight=1.0

Apparently, the training file does not have the information on the charges' weight besides the charges' key. Does it mean charges are still not implemented during the training process?

Thank you.

**Comment by ilyes319** [comment]
One can not train on charges weight, it will come soon.
